src/platform/plugins/shared/discover/public/embeddable/utils/serialization_utils.ts

```diff
@@ -136,6 +136,7 @@ export const serializeState = ({
     rawState: {
       ...serializeTitles(),
       ...serializeTimeRange(),
+      ...serializeDynamicActions?.(),
```

Comment:
This is a bug in main, where drilldowns are never serialized to by-value discover sessions.

Reply:
Oh, I thought I'd tested in main, but maybe I was using a by-ref embeddable. Thanks for fixing.

Reply:
It is a little different in main. In main, the unsaved changes use runtime state so the unsaved changes badge shows. However, upon saving, the drill down state is not spread in serialized state. I will open an issue and fix in main so it can be backported to other branches.